packages/build-utils/src/prerender.ts

```diff
@@ -52,6 +55,17 @@ export class Prerender {
     }
     this.group = group;
 
+    if (passQuery === true) {
+      this.passQuery = true;
+    } else if (
+      typeof passQuery !== 'boolean' &&
+      typeof passQuery !== 'undefined'
+    ) {
+      throw new Error(
```

Comment:
This should throw a `TypeError`.

It would be rad to have a test that passes in something that isn't `undefined`, `true`, or `false`.

Reply:
Hmm not sure if we should use `TypeError` specifically as that deviates from the existing checks, can add an assertion to make sure it throws as expected though.

Reply:
Added in https://github.com/vercel/vercel/pull/9388/commits/47f939fcb599d76c248b46d321ea58993db2dd0f

Reply:
A `TypeError` is specifically for type errors which is exactly what this code is testing for.

For example, checking that the `bypassToken` has a length < 32 is not a `TypeError` because it's validating the variable contents, not its type.

Technically `fallback` and `initialStatus` should also throw `TypeError`. At the end of the day, `TypeError` is still an instanceof `Error` and shouldn't cause an issue.

Reply:
Would prefer to keep changes like this in a separate PR to reduce risk as there can be unexpected side effects from `Error` or `TypeError` being serialized/deserialized. 

Additional risk is being assumed as we aren't currently using `TypeError` anywhere in `build-utils` so no prior guarantees this just works throughout the pipeline.